python-test-samples/apigw-sqs-lambda-sqs/tests/integration/test_api_gatway.py

```diff
@@ -0,0 +1,305 @@
+        logging.info("Setup APIGatewayURL_Outbox: %s", self.api_endpoint_outbox)
+        logging.info("Setup SQSInputQueue: %s", self.sqs_input)
+        logging.info("Setup SQSOutputQueue: %s", self.sqs_output)
+        logging.info("Setup SQSInputQueueDLQ: %s", self.sqs_input_dlq)
+        logging.info("Setup SQSOutputQueueDLQ: %s", self.sqs_output_dlq)
+        logging.info("Setup DynamoDBTable: %s", self.dynamodb_table_name)
+
+        # Seed the DynamoDB Table with Test Data
+        self.dynamodb_resource = boto3.resource("dynamodb", region_name = self.aws_region)
+        self.dynamodb_table = self.dynamodb_resource.Table(name=self.dynamodb_table_name)
+
+        # The following item, just a sample written dircetly to dynamodb
+        # for debug purpuses, and will be deleted in teardown phase
+        # self.dynamodb_table.put_item(Item={"id": "TEST000" + self.id_postfix,
+        #                               "Queue_Name": "NAME#",
+        #                               "Message": self.test_time + " Init DB Record" })
+
+    def teardown_class(self) -> None:
+        """
+        # For tear-down, remove any data injected for the tests
+        # purge each queue in case they have messages inside them
+        # since purge can be done once every 60 sec, we start with 60 sec sleep
+        """
+        logging.info("Teardown Phase...")
+
+        # Take particular care to ensure these values are unique and identifiable as TEST data.
+        logging.info("Teardown DynamoDB...")
+
+        for test_id in ["TEST000", "TEST001" ,"TEST002", "TEST003"]:
+            message_id = test_id + self.id_postfix
+            id_items = self.dynamodb_table.query(
+                KeyConditionExpression=Key('id').eq(message_id))
+
+            if "Items" in id_items:
+                for item in id_items["Items"]:
+                    self.dynamodb_table.delete_item(Key={"id":item["id"]})
+
+        # cleaning SQS queues
+        logging.info("Teardown SQS...")
+
+        client = boto3.client("sqs")
+        response = {}
+
+        response = client.get_queue_attributes(
+            QueueUrl=self.sqs_input,
+            AttributeNames=['ApproximateNumberOfMessages']
+        )
+
+        message_count = int(response['Attributes']
+                            ['ApproximateNumberOfMessages'])
+        if message_count > 0:
+            logging.info("Cleaning input queue")
+            client.purge_queue(QueueUrl=self.sqs_input)  # purging input queue
+
+        response = client.get_queue_attributes(
+            QueueUrl=self.sqs_output,
+            AttributeNames=['ApproximateNumberOfMessages']
+        )
+
+        message_count = int(response['Attributes']
+                            ['ApproximateNumberOfMessages'])
+        if message_count > 0:
+            logging.info("Cleaning output queue")       # purging output queue
+            client.purge_queue(QueueUrl=self.sqs_output)
```

Comment:
This will purge all messages, not just test messages?

Reply:
removed the QSQ purge not relevant any more